src/elfpy/markets.py

```diff
@@ -216,50 +294,71 @@ def update_spot_price(self):
+                output_without_fee,
+                fee,
+            ) = trade_results
+        return without_fee_or_slippage, output_with_fee, output_without_fee, fee,
 
     def _open_short(self, trade_details):
         """
         take trade spec & turn it into trade details
         compute wallet update spec with specific details
             will be conditional on the pricing model
         """
-        trade_results = self.pricing_model.calc_out_given_in(
-            trade_details.trade_amount,
-            trade_details.share_reserves,
-            trade_details.bond_reserves,
-            trade_details.token_out,
-            trade_details.fee_percent,
-            trade_details.stretched_time_remaining,
-            trade_details.init_share_price,
-            trade_details.share_price,
+        without_fee_or_slippage, output_with_fee, output_without_fee, fee = self.swap(trade_details)
+        print(f"opening short: {without_fee_or_slippage, output_with_fee, output_without_fee, fee}")
+        market_deltas = MarketDeltas( # write out explicit signs, so it's clear what's happening
+            d_base_asset                = - output_with_fee,
+            d_token_asset               = + trade_details.trade_amount,
+            d_bond_buffer               = + trade_details.trade_amount,
+            d_share_fee                 = + fee / trade_details.share_price,
+            d_share_fee_history         = {trade_details.mint_time: fee / trade_details.share_price},
+            d_base_asset_slippage       = + abs(without_fee_or_slippage - output_without_fee),
+            d_base_asset_orders         = + 1,
+            d_base_asset_volume         = + output_with_fee,
         )
-        (
-            without_fee_or_slippage,
-            output_with_fee,
-            output_without_fee,
-            fee,
-        ) = trade_results
-        market_deltas = {
-            "d_base_asset": -output_with_fee,
-            "d_token_asset": trade_details.trade_amount,
-            "d_base_asset_slippage": abs(without_fee_or_slippage - output_without_fee),
-            "d_token_asset_slippage": 0,
-            "d_base_asset_fee": fee,
-            "d_token_asset_fee": 0,
-            "d_base_asset_orders": 1,
-            "d_token_asset_orders": 0,
-            "d_base_asset_volume": output_with_fee,
-            "d_token_asset_volume": 0,
-        }
         # TODO: _in_protocol values should be managed by pricing_model and referenced by user
         max_loss = trade_details.trade_amount - output_with_fee
-        wallet_deltas = {
-            "base_in_wallet": -1 * max_loss,
-            "base_in_protocol": [trade_details.mint_time, max_loss],
-            "token_in_wallet": None,
-            "token_in_protocol": [trade_details.mint_time, trade_details.trade_amount],
-            "fee": [trade_details.mint_time, fee],
-        }
+        wallet_deltas = AgentWallet( # write out explicit signs, so it's clear what's happening
```

Comment:
I don't think this comment is really necessary, it's clear what you're doing below.